public/app/plugins/datasource/loki/components/LokiExploreQueryEditor.tsx

```diff
@@ -0,0 +1,87 @@
+// Libraries
+import React, { memo, useState } from 'react';
+import _ from 'lodash';
+
+// Types
+import { AbsoluteTimeRange, ExploreQueryFieldProps } from '@grafana/data';
+import { LokiDatasource } from '../datasource';
+import { LokiQuery, LokiOptions } from '../types';
+import { LokiQueryField } from './LokiQueryField';
+import { useLokiSyntax } from './useLokiSyntax';
+import { FormLabel } from '@grafana/ui';
+
+type Props = ExploreQueryFieldProps<LokiDatasource, LokiQuery, LokiOptions>;
+
+export const LokiQueryEditor = memo(function LokiQueryEditor(props: Props) {
+  const { query, data, datasource, exploreMode, history, onChange, onRunQuery } = props;
+
+  let absolute: AbsoluteTimeRange;
+  if (data && !_.isEmpty(data.request)) {
+    const { range } = data.request;
+
+    absolute = {
+      from: range.from.valueOf(),
+      to: range.to.valueOf(),
+    };
+  } else {
+    absolute = {
+      from: Date.now() - 10000,
+      to: Date.now(),
+    };
+  }
+
+  const [maxLines, setMaxLines] = useState('');
+
+  const { isSyntaxReady, setActiveOption, refreshLabels, ...syntaxProps } = useLokiSyntax(
```

Comment:
@kaydelaney we should probably refactor the props of the loki field because right now these syntax props are required but at the same time they are actually created in `LokiQueryField` and so we call `useLokiSyntax` two times.